src/closed.jl

```diff
@@ -0,0 +1,44 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
+"""
+immutable ClosedSet{T}
+	left::T
+	right::T
+end
+
+function ClosedSet(left, right)
+	(I_left, I_right) = promote(left, right)
+	ClosedSet(I_left, I_right)
+end
+
+show(io::IO, I::ClosedSet) = print(io, "[", I.left, ",", I.right, "]")
+
+length(I::ClosedSet) = I.right - I.left + 1
+
+in(v, I::ClosedSet) = I.left <= v <= I.right
+
+isempty(A::ClosedSet) = A.left > A.right
+
+isequal(A::ClosedSet, B::ClosedSet) = (A.left == B.left && A.right == B.right) || isempty(A) && isempty(B)
+
+==(A::ClosedSet, B::ClosedSet) = isequal(A, B)
+
+function intersect(A::ClosedSet, B::ClosedSet)
+	left = max(A.left, B.left)
+	right = min(A.right, B.right)
+	left, right = promote(left, right)
+	ClosedSet(left, right)
+end
+
+function union(A::ClosedSet, B::ClosedSet)
+	left = min(A.left, B.left)
+	right = max(A.right, B.right)
+	left, right = promote(left, right)
+	ClosedSet(left, right)
+end
+
+issubset(A::ClosedSet, B::ClosedSet) =  isempty(A) || (A.left in B && A.right in B)
```

Comment:
This becomes faster if `|` and `&` are used. (First benchmark is for the original one with `||` and `&&`)- 

```julia
julia> @benchmark issubset(A, B)
julia> @benchmark issubset(A, B)
BenchmarkTools.Trial: 
  samples:          10000
  evals/sample:     995
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  0.00 bytes
  allocs estimate:  0
  minimum time:     27.00 ns (0.00% GC)
  median time:      27.00 ns (0.00% GC)
  mean time:        27.70 ns (0.00% GC)
  maximum time:     71.00 ns (0.00% GC)

BenchmarkTools.Trial: 
  samples:          10000
  evals/sample:     997
  time tolerance:   5.00%
  memory tolerance: 1.00%
  memory estimate:  0.00 bytes
  allocs estimate:  0
  minimum time:     22.00 ns (0.00% GC)
  median time:      22.00 ns (0.00% GC)
  mean time:        22.40 ns (0.00% GC)
  maximum time:     49.00 ns (0.00% GC)
```